packages/service-clients/azure-client/src/AzureClient.ts

```diff
@@ -170,47 +167,31 @@ export class AzureClient {
 			encodeURIComponent(getTenantId(this.properties.connection)),
 		);
 		url.searchParams.append("containerId", encodeURIComponent(id));
-		const sourceContainer = await loader.resolve({ url: url.href });
-
-		if (sourceContainer.resolvedUrl === undefined) {
-			throw new Error("Source container cannot resolve URL.");
-		}
-
-		const documentService = await this.documentServiceFactory.createDocumentService(
-			sourceContainer.resolvedUrl,
-		);
-		const storage = await documentService.connectToStorage();
-		const handle = {
-			type: SummaryType.Handle,
-			handleType: SummaryType.Tree,
-			handle: version?.id ?? "latest",
-		};
-		const tree = await storage.downloadSummary(handle);
-
-		const container = await loader.rehydrateDetachedContainerFromSnapshot(JSON.stringify(tree));
-
-		const fluidContainer = await this.createFluidContainer<TContainerSchema>(
+		const container = await loader.resolve({ url: url.href });
+		const rootDataObject = await this.getContainerEntryPoint(container);
+		const fluidContainer = createFluidContainer<TContainerSchema>({
 			container,
-			this.properties.connection,
-		);
+			rootDataObject,
+		});
 		const services = this.getContainerServices(container);
 		return { container: fluidContainer, services };
 	}
 
 	/**
-	 * Accesses the existing container given its unique ID in the Azure Fluid Relay.
+	 * Load a specific version of a container for viewing only.
 	 * @typeparam TContainerSchema - Used to infer the the type of 'initialObjects' in the returned container.
 	 * (normally not explicitly specified.)
-	 * @param id - Unique ID of the container in Azure Fluid Relay.
+	 * @param id - Unique ID of the source container in Azure Fluid Relay.
 	 * @param containerSchema - Container schema used to access data objects in the container.
-	 * @returns Existing container instance along with associated services.
+	 * @param version - Unique version of the source container in Azure Fluid Relay.
+	 * @returns Loaded container instance at the specified version.
 	 */
-	public async getContainer<TContainerSchema extends ContainerSchema>(
+	public async viewContainerVersion<TContainerSchema extends ContainerSchema>(
 		id: string,
 		containerSchema: TContainerSchema,
+		version: AzureContainerVersion,
```

Comment:
Is "version" being used at all?

Reply:
Good call, I accidentally broke it when switching to the new API in the last commit.  Fixed in latest.